spec/services/github/contributions/collect_spec.rb

```diff
@@ -21,11 +22,47 @@
     end
 
     context 'when there are no repositories in database' do
-      let(:company) { build_stubbed(:company) }
+      it { is_expected.to be_empty }
+    end
+
+    context 'when there are no engineers in database' do
+      let(:repositories) { double(pluck: [[1, 'http://example.com']]) }
 
+      before do
+        allow(company).to receive(:repositories).and_return(repositories)
+      end
+      
       it { is_expected.to be_empty }
     end
 
-    # TODO: Add specs to cover all use cases
+    context 'when Github API is not able to respond' do
+      let(:repositories) { double(pluck: [[1, 'http://example.com']]) }
+      let(:engineers) { double(active: double(pluck: [['github_user', 1]])) }
+      let(:users) { double(engineer: engineers) }
+
+      before do
+        allow(company).to receive(:repositories).and_return(repositories)
+        allow(company).to receive(:users).and_return(users)
+        allow(client).to receive(:search) { raise 'any error' }
+      end
+
+      it { is_expected.to be_empty }
+    end
+
+    context 'when Pull Requests are fetched from GitHub API' do
+      let(:repositories) { double(pluck: [[1, 'http://example.com']]) }
+      let(:engineers) { double(active: double(pluck: [['github_user', 1]])) }
+      let(:users) { double(engineer: engineers) }
+      let(:pull_requests) { double(items: [ double(created_at: '2022-01-01', html_url: 'http://example.com/owner/name/pull/123', user: double(login: 'github_user')) ]) }
+      let(:client_search) { double(issues: pull_requests) }
+
+      before do
+        allow(company).to receive(:repositories).and_return(repositories)
+        allow(company).to receive(:users).and_return(users)
+        allow(client).to receive(:search).and_return(client_search)
+      end
+
+      it { is_expected.not_to be_empty }
```

Comment:
IMO this spec is weak because anything that responds to `empty?` method and returns true, can turn this spec into a false green. Can we ensure the returned object here?

`it { is_expected.to eq [something] }` or even `it { is_expected.to include something }`